services/api/service.go

```diff
@@ -830,6 +888,50 @@ func (api *RelayAPI) handleGetPayload(w http.ResponseWriter, req *http.Request)
 		if err != nil {
 			log.WithError(err).Error("failed to increment builder-stats after getPayload")
 		}
+
+		builderStatus, err := api.redis.GetBlockBuilderStatus(bidTrace.BuilderPubkey.String())
+		if err != nil {
+			log.WithError(err).Error("failed to get block builder status")
+		}
+
+		// Check if the block was valid in the optimistic case.
+		if builderStatus == common.Optimistic {
+			simErr := api.simulateBlock(blockSimOptions{
```

Comment:
Would it be fair to say that the block would be simulated twice?

Reply:
yes! we simulate all the optimistic blocks we receive, but the optimistic block processor doesn't make any decisions about refunds. that happens here, in the getPayload handler. does that seem reasonable? the only blocks that get simulated twice are those that are actually published.